### PR TITLE
Add MLX op handler for aten.bitwise_or

### LINKED ISSUE #18926: Good First Issue: Add MLX Op Handler for `aten.bitwise_or`
### 🚀 The feature, motivation and pitch

# Good First Issue: Add MLX Op Handler for `aten.bitwise_or`

## Summary

Add support for `aten.bitwise_or` in the MLX delegate. This op performs element-wise bitwise OR and is needed for bit manipulation, flags/masks, and low-level data processing.

## Background

The MLX delegate currently has no handler for `bitwise_or`. MLX has native support via `mlx::core::bitwise_or` which works for both boolean and integer tensors.

### Approach: New schema node + runtime

Add a `BitwiseOrNode` to handle this via MLX's native `bitwise_or`.

## Steps

1. **Add node to `backends/mlx/serialization/schema.fbs`**
   ```flatbuffers
   table BitwiseOrNode {
     a: Tid;
     b: Tid;
     out: Tid;
   }
   ```
   Add `BitwiseOrNode` to the `OpNode` union (append only, do not reorder).

2. **Regenerate serialization code**
   ```bash
   python backends/mlx/serialization/generate.py
   ```

3. **Add C++ runtime exec function in `backends/mlx/runtime/MLXInterpreter.h`**
   ```cpp
   inline void exec_bitwise_or(
       const BitwiseOrNode& n, ExecutionState& st, StreamOrDevice s) {
     auto a = st.get_tensor(n.a());
     auto b = st.get_tensor(n.b());
     auto out = mlx::core::bitwise_or(a, b, s);
     st.set_tensor(n.out(), out);
   }
   ```

4. **Add handler in `backends/mlx/ops.py`**
   
   Add to the `_BINARY_OPS` table:

   ```python
   # In _BINARY_OPS list, add:
   ([torch.ops.aten.bitwise_or.Tensor, torch.ops.aten.bitwise_or.Scalar], BitwiseOrNode, "aten.bitwise_or", True),
   ```

5. **Add test in `backends/mlx/test/test_ops.py`**
   
   Use the `_BINARY_OP_TESTS` table:

   ```python
   # Add to _BINARY_OP_TESTS list:
   {"op_name": "bitwise_or_bool", "op_fn": torch.bitwise_or, 
    "shapes": _SHAPES_3, "dtypes": [torch.bool], 
    "input_fn_a": _bool_input_fn(), "input_fn_b": _bool_input_fn()},
   {"op_name": "bitwise_or_int", "op_fn": torch.bitwise_or, 
    "shapes": _SHAPES_3, "dtypes": [torch.int32, torch.int64], 
    "input_fn_a": _int_input_fn(0, 256), "input_fn_b": _int_input_fn(0, 256)},
   ```

### Running tests

```bash
python -m executorch.backends.mlx.test.run_all_tests -k bitwise_or
```

## References

- MLX C++: `array bitwise_or(const array &a, const array &b, StreamOrDevice s = {})`
- PyTorch signature: `bitwise_or(Tensor self, Tensor other) -> Tensor`
- Supported dtypes: `int8`, `int16`, `int32`, `int64`, `uint8`, `bool`

### Alternatives

_No response_

### Additional context

_No response_

### RFC (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Jah-yee**
Good day

Add MLX op handler for aten.bitwise_or (issue #18926).

## Changes
- Add BitwiseOrNode table to schema.fbs (following bitwise_xor pattern)
- Add BitwiseOrNode to OpNode union
- Regenerate serialization code
- Add exec_bitwise_or function to MLXInterpreter.h
- Add BITWISE_OR case handler to interpreter switch
- Add aten.bitwise_or to _BINARY_OPS in ops.py
- Add bitwise_or test case to test_ops.py

## Testing
```bash
python -m executorch.backends.mlx.test.run_all_tests -k bitwise_or
```

## References
- Fixes: pytorch/executorch#18926

Thank you for your attention. If there are any issues or suggestions, please leave a comment and I will address them promptly.

Warmly,
RoomWithOutRoof

cc @metascroy

### CODE REVIEW

**Comment by nil-is-all** [comment]
@claude review

File: backends/mlx/test/test_ops.py [hunk 1]
```diff
@@ -4222,6 +4205,7 @@ def create_model(self) -> nn.Module:
     # logical
     {"op_name": "logical_and",   "op_fn": torch.logical_and, "shapes": [(2, 3, 4), (10,), (4, 8)], "dtypes": [torch.bool], "input_fn_a": _bool_input_fn(), "input_fn_b": _bool_input_fn()},
     {"op_name": "logical_or",    "op_fn": torch.logical_or,  "shapes": [(2, 3, 4), (10,), (4, 8)], "dtypes": [torch.bool], "input_fn_a": _bool_input_fn(), "input_fn_b": _bool_input_fn()},
+    {"op_name": "bitwise_or",    "op_fn": torch.bitwise_or,  "shapes": [(2, 3, 4), (10,), (4, 8)], "dtypes": [torch.int32, torch.int64], "input_fn_a": _int_input_fn(-100, 100), "input_fn_b": _int_input_fn(-100, 100)},
```
Comment (metascroy) [inline backends/mlx/test/test_ops.py:4208]: @Jah-yee do these pass? I'm a bit surprised if they do, because I don't see a handler in backends/mlx/ops.py?

**Comment by metascroy** [comment]
Hi @Jah-yee, thanks for the PR! A couple notes:

* I think we need a handler for the new op in backends/mlx/ops.py. See logical_or/and as examples.
* Do not check in generated code (MLXLoader.cpp, MLXLoader.h, etc)

**Comment by nil-is-all** [comment]
Hi @Jah-yee, could you address the review?
